src/components/v5/shared/Navigation/Sidebar/partials/SidebarRouteItem/SidebarRouteItem.tsx

```diff
@@ -0,0 +1,137 @@
+import { CaretDown } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import { AnimatePresence, motion } from 'framer-motion';
+import { noop } from 'lodash';
+import React, { useState } from 'react';
+import { useMatch, useNavigate } from 'react-router-dom';
+
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { usePageLayoutContext } from '~context/PageLayoutContext/PageLayoutContext.ts';
+import { useTablet } from '~hooks';
+import { formatText } from '~utils/intl.ts';
+import buttonClasses from '~v5/shared/Button/Button.styles.ts';
+
+import {
+  sidebarButtonClass,
+  sidebarButtonIconClass,
+  sidebarButtonTextClass,
+} from '../../sidebar.styles';
+
+import { motionVariants } from './consts.ts';
+import { type SidebarRouteItemProps } from './types.ts';
+
+const displayName = 'v5.shared.Navigation.Sidebar.partials.SidebarRouteItem';
+
+const SidebarRouteItem: React.FC<SidebarRouteItemProps> = ({
+  path,
+  translation,
+  icon: Icon,
+  routeType,
+  subItems,
+  onClick,
+}) => {
+  const { colony } = useColonyContext();
+
+  const navigate = useNavigate();
+
+  const matchingRoute = useMatch(`${colony.name}/${path}`);
```

Comment:
Maybe we can use a constant ```const isRouteActive = !!useMatch(`${colony.name}/${path}`);``` ?

Reply:
This is a great shout @mmioana! Is it all right for you if I amended this on a separate PR? I have a draft issue for refactoring because otherwise, my soul will never find peace since I have a tendency to just refactor everything I come across 😂 

https://github.com/orgs/JoinColony/projects/70?pane=issue&itemId=73741708